File: mahjong_utils/models/wind.py
from enum import Enum

from mahjong_utils.models.tile import tile, Tile
from mahjong_utils.models.tile_type import TileType


class Wind(int, Enum):
    """
    风
    """

    east = 0
    south = 1
    west = 2
    north = 3

    @property
    def tile(self) -> Tile:
        return tile(TileType.Z, self + 1)


__all__ = ("Wind",)

File: mahjong-model/src/commonMain/kotlin/mahjongutils/models/Wind.kt
package mahjongutils.models

/**
 * 风（东、南、西、北）
 */
enum class Wind(
    /**
     * 对应的风牌
     */
    val tile: Tile
) {
    East(Tile.get(TileType.Z, 1)),
    South(Tile.get(TileType.Z, 2)),
    West(Tile.get(TileType.Z, 3)),
    North(Tile.get(TileType.Z, 4)),
}